src/models/registration/pointnet2.py

```
@@ -91,4 +91,5 @@ def backward(self):
         """Calculate losses, gradients, and update network weights; called in every training iteration"""
         # caculate the intermediate results if necessary; here self.output has been computed during function <forward>
         # calculate loss given the input and intermediate results
-        self.loss.backward()  # calculate gradients of network G w.r.t. loss_G
+        if hasattr(self, "loss") and self.loss:
```

Are you sure you can do that ? Won't it trigger an error if you don't have loss in self.

---

You can see[here](https://github.com/nicolas-chaulet/deeppointcloud-benchmarks/blob/74784e2b37f2c7c24bc819eee9deef00a4152dda/src/models/base_architectures/backbone.py#L101), where we add the loss. If in the config, no loss has been provided, it will be None. You're right. We should throw an exception if the loss is None (The miner can be None)

---

Nope, it is tested :)

---

The only way the loss does not exists is if the forward functions stops earlier (there is a return half way through). I'll leave the hasattr but remove the second check so that it throws an exception